### PR TITLE
Better time_range vs time_array handling in UVCal

### LINKED ISSUE #851: time_array and time_range are problematic in UVCal
time_array should be used, but time_range should be replaced by an integration_length-like array that defines what range of times the solution applies to.  This will require some work on uvcalibrate to interpret sensibly.

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

Improve the way `time_range` is handled in UVCal. It exists to support calibration approaches that solve for cal solutions over a range of times rather than per integration. This PR makes the following changes:
- Deprecate having both a `time_array` and a `time_range` defined.
- Make `time_range` be 2D with a shape of (Ntimes, 2) where the first axis gives the number of different time solutions and the second axis gives the start/stop times for those solutions. This allows different calibration solutions that apply over ranges of times to be properly combined in a single object.
- Add a `get_time_array` method that either returns the mean of the start and stop time for each time range or the time_array (if there's a time_array and no time_range)

**Note**: this is actually a breaking change in the following ways:
- It changes the shape of the `time_range` in the object.
- It makes it so that `time_range` is not defined on an object read from a calfits file with more than one time, even if the file has a `time_range` entry in the header. This is because the header `time_range` is a single range, it doesn't have a length Ntimes axis, so it doesn't have the right shape for the new `time_range` shape.
- If a calfits file is read in with one time and it has a `time_range` entry in the header, only the `time_range` will be defined, not the `time_array`. This could be changed but it would result in many deprecation warnings because having both a `time_range` and a `time_array` is deprecated. We could add a parameter to `read_calfits` to only have it define one or the other.
- If and fhd cal file is read in (which by definition has only one time), only the `time_range` will be defined, not the `time_array`. This could be changed but it would result in many deprecation warnings because having both a `time_range` and a `time_array` is deprecated. We could add a parameter to the `read_fhd` method to only have it define one or the other.

I see how to fix the third  & fourth items above in a way that supports deprecation, but I don't see a path for allowing for a deprecation period for the first two items. So we decided to push this into version 3.0.

## Motivation and Context

Closes #851 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Breaking change checklist:
- [x] I have updated the docstrings associated with my change using the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).
- [x] I have updated the tutorial to reflect my changes (if appropriate).
- [x] My change includes backwards compatibility and deprecation warnings (if possible).
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/main/CHANGELOG.md).

### CODE REVIEW

File: pyuvdata/uvcal/uvcal.py [hunk 1]
```diff
@@ -238,36 +294,70 @@ def __init__(self):
...
+        self._lst_range = uvp.UVParameter(
+            "lst_range",
+            description=desc,
+            form=("Ntimes", 2),
+            expected_type=float,
+            tols=uvutils.RADIAN_TOL,
+            required=False,
+        )
+
+        # TODO:
```
Comment (jpober) [inline pyuvdata/uvcal/uvcal.py:357]: Should we turn these into issues?

**Review by jpober — DISMISSED** [review]

**Comment by jpober** [comment]
I think we expect the hera_cal and hera_qm builds to fail, but is the build on pyuvdata failing expected?  Is it because this is a breaking change?

**Comment by bhazelton** [comment]
That third failure is just the roll up of the set of external tests, so it fails if any of them do. I don't know why it's listed separately.

File: pyuvdata/uvcal/uvcal.py (hunk 1 above)
Comment (bhazelton) [reply pyuvdata/uvcal/uvcal.py:357]: Actually, I think those are cruft. The first one is done and I think we sorted it out reasonably without the other package.

**Comment by bhazelton** [comment]
@jpober can you please re-approve? I removed the stale TODOs

**Review by jpober — APPROVED** [review]

**Pull request merged**